ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageUploader.scala

```diff
@@ -0,0 +1,127 @@
+package com.daml.ledger.participant.state.kvutils.committer
+
+import com.daml.daml_lf_dev.DamlLf.Archive
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.Invalid
+import com.daml.ledger.participant.state.kvutils.committer.Committer.StepInfo
+import com.daml.lf.archive.Decode
+import com.daml.lf.data.Ref
+import com.daml.lf.engine.Engine
+import com.daml.lf.language.Ast
+import com.daml.metrics.Metrics
+
+import scala.collection.JavaConverters._
+
+/*
+ * This PackagesCommitter performs a minimal package validation before commit and
+ * upload the packages in the engine in a background process.
+ *
+ * This is design for kvutils integrations that cannot long-running submissions.
+ * When using this committer, packages on the ledger cannot be trusted and should be
+ * validated each time uploaded to the engine.
+ */
+private[committer] class PackageUploader(
```

Comment:
Sorry for not bringing this up in previous review round, but the naming here is also confusing. Could you name these so that it's clear to the reader that these are different "PackageCommitter" variants? E.g. NonValidatingPackageCommitter, HashValidatingPackageCommitter and maybe FullValidatingPackageCommitter (or better names if you can think of them). This way it's clear at a glance what these are.

Reply:
I agree with Jussi.